bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go

```diff
@@ -160,6 +162,9 @@ func (r *KubeadmConfigReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		log.Error(err, "Failed to get config")
 		return ctrl.Result{}, err
 	}
+	if r.Filter != nil && !r.Filter.Filter(config) {
```

Comment:
Filters should be applied before actually entering in the reconcile loop (e.g. in while setting up controllers)

Reply:
But then how can we filter out resources that are triggered by another kind of resource?